tests/tt_metal/tt_metal/unit_tests_fast_dispatch/command_queue/test_remote_dispatch_tunneling.cpp

```diff
@@ -0,0 +1,88 @@
+    for (unsigned int id = 0; id < devices_.size(); id++) {
+        auto device = devices_.at(id);
+        if (device->is_mmio_capable()) {
+            continue;
+        }
+
+        CommandQueue &remote_cq = detail::GetCommandQueue(device);
+
+        chip_id_t mmio_device_id = tt::Cluster::instance().get_associated_mmio_device(device->id());
+        uint16_t channel = tt::Cluster::instance().get_assigned_channel_for_device(device->id());
+        uint8_t num_hw_cqs = device->num_hw_cqs();
+        ASSERT_EQ(num_hw_cqs, 1) << "Expected this test to be run with single HW CQ test suite!";
+        uint8_t cq_id = 0;
+
+        uint32_t num_pages = 1;
+        uint32_t page_size = 2048;
+        uint32_t buff_size = num_pages * page_size;
+        Buffer bufa(device, buff_size, page_size, BufferType::DRAM);
+
+        std::vector<uint32_t> src(buff_size / sizeof(uint32_t), 0);
+        for (uint32_t i = 0; i < src.size(); i++) {
+            src.at(i) = i;
+        }
+
+        EnqueueWriteBuffer(remote_cq, bufa, src.data(), false);
+
+        uint32_t num_bytes_in_cmd_header = DeviceCommand::NUM_ENTRIES_IN_COMMAND_HEADER * sizeof(uint32_t);
+
+        std::vector<uint32_t> cq_header(DeviceCommand::NUM_ENTRIES_IN_COMMAND_HEADER);
+        tt::Cluster::instance().read_sysmem(cq_header.data(), num_bytes_in_cmd_header, CQ_START, mmio_device_id, channel);
+
+        tt_cxy_pair issue_q_reader_location = dispatch_core_manager::get(num_hw_cqs).issue_queue_reader_core(device->id(), channel, cq_id);
+        CoreCoord issue_q_physical_core = tt::get_physical_core_coordinate(issue_q_reader_location, CoreType::WORKER);
+        std::vector<uint32_t> issue_q_reader_header(DeviceCommand::NUM_ENTRIES_IN_COMMAND_HEADER);
+        tt::Cluster::instance().read_core(issue_q_reader_header.data(), num_bytes_in_cmd_header, tt_cxy_pair(issue_q_reader_location.chip, issue_q_physical_core), L1_UNRESERVED_BASE);
+        EXPECT_EQ(cq_header, issue_q_reader_header) << "Remote issue queue reader did not read in expected command!";
+
+        tt_cxy_pair src_eth_router_location = tt::Cluster::instance().get_eth_core_for_dispatch_core(issue_q_reader_location, EthRouterMode::FD_SRC, device->id());
+        CoreCoord physical_src_eth_router = tt::get_physical_core_coordinate(src_eth_router_location, CoreType::ETH);
+        std::vector<uint32_t> source_router_header(DeviceCommand::NUM_ENTRIES_IN_COMMAND_HEADER);
+        tt::Cluster::instance().read_core(source_router_header.data(), num_bytes_in_cmd_header, tt_cxy_pair(src_eth_router_location.chip, physical_src_eth_router), eth_l1_mem::address_map::ERISC_APP_RESERVED_BASE);
+        EXPECT_EQ(cq_header, source_router_header) << "Ethernet router on local chip (src) did not receive expected command!";
+
+        // Need ethernet cores ot get out of FD mode to do host read backs from remote device
+        tt::Cluster::instance().set_internal_routing_info_for_ethernet_cores(false);
```

Comment:
@aliuTT  I think you mentioned that i don't need to do this to do the host issued remote reads below, but I was seeing a hang without it

Reply:
You're reading from remote device below no?

Reply:
yes

Reply:
I probably meant to say you don't need set(false) here if you only read back from the mmio device. The remote read will go through your ethernet cores, so the `false` yields those cores